File: src/main/java/cu/suitetecsa/sdk/nauta/utils/ExceptionFactory.java
package cu.suitetecsa.sdk.nauta.utils;

/**
 * Esta interfaz define una factoría de excepciones que proporciona un método para crear una excepción
 * con un mensaje dado.
 */
public interface ExceptionFactory<T extends Throwable>  {
    /**
     * Crea una excepción con el mensaje dado.
     *
     * @param message El mensaje de la excepción.
     * @return Una instancia de excepción creada.
     */
    T createException(String message);
}

File: src/main/kotlin/io/github/suitetecsa/sdk/nauta/utils/ExceptionFactory.kt
package io.github.suitetecsa.sdk.nauta.utils

/**
 * Esta interfaz define una factoría de excepciones que proporciona un método para crear una excepción
 * con un mensaje dado.
 */
fun interface ExceptionFactory<T : Throwable> {
    /**
     * Crea una excepción con el mensaje dado.
     *
     * @param message El mensaje de la excepción.
     * @return Una instancia de excepción creada.
     */
    fun createException(message: String): T
}
